src/lib/y2partitioner/widgets/md_add_button.rb

```diff
@@ -0,0 +1,49 @@
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "cwm"
+require "y2partitioner/actions/add_md"
+
+module Y2Partitioner
+  module Widgets
+    # Button for openng a wizard to add a new MD array
+    class MdAddButton < CWM::PushButton
```

Comment:
Why wasn't this turned into a `DeviceButton`  like `LvmLvAddButton` and `PartitionAddButton`?

Reply:
As commented, `LvmVgAddButton` and `MdAddButton` do not work over a specific device. The class `DeviceButton` is meant to act over a device.